shoov/modules/custom/shoov_screenshot/shoov_screenshot.module

```diff
@@ -126,3 +126,50 @@ function shoov_screenshot_add_tag_to_vocabulary($term_name, $vocabulary_id) {
 
   return $tid;
 }
+
+/**
+ * Create a hash code from files content.
+ *
+ * @param $files
+ *  List of uploaded files info.
+ * @param $ui_build_id
+ *  The UI Build node ID.
+ *
+ * @return string
+ *  Returns the hash code.
+ */
+function shoov_screenshot_create_hash($files, $ui_build_id) {
+  $hash = array();
+  foreach ($files as $file) {
+    $image = file_load($file['id']);
+    $hash[] = $file['label'] . ':' . file_get_contents($image->uri) . ';';
+  }
+  $hash[] = $ui_build_id;
+
+  $hash = md5(join(';', $hash));
+  return $hash;
+}
+
+/**
+ * Get the screenshot ID with the same hash code if exists.
+ *
+ * @param $hash
+ *  The hash code.
+ *
+ * @return array|bool
```

Comment:
there would be only a single one.

For the existing duplicates we will run a script that will delete them.